test/benchmarks/mongoBench/runner.js

```diff
@@ -162,12 +163,17 @@ class Runner {
     const minExecutionCount = this.minExecutionCount;
     const minExecutionTime = this.minExecutionTime;
     const maxExecutionTime = this.maxExecutionTime;
+    const maxExecutionAttempts = this.minExecutionCount;
     let time = performance.now() - start;
     let count = 1;
 
     const taskTimer = benchmark._taskType === 'sync' ? timeSyncTask : timeAsyncTask;
 
-    while (time < maxExecutionTime && (time < minExecutionTime || count < minExecutionCount)) {
+    while (
+      time < maxExecutionTime &&
+      (time < minExecutionTime || count < minExecutionCount) &&
+      count < maxExecutionAttempts
+    ) {
```

Comment:
```js
 _errorHandler(error) {
    let currentError = error;
    while (currentError) {
      this.reporter(
        `${currentError !== error ? 'Caused by' : 'Error'}: ${currentError.name} - ${currentError.message} - ${currentError.stack}`
      );
      currentError = currentError.cause;
    }
    throw error;
  }
```

I don't feel too strongly about the print logic, it was just to remove all those ids from the bulk error, but can we make errors throw now? I don't think we should have the code "tolerate" errors and continue and ruin the entire calculation with `NaN` anymore.